tests/cupy_tests/padding_tests/test_pad.py

```diff
@@ -263,35 +265,56 @@ def test_pad_special(self, xp):
 
 @testing.parameterize(
     {'array': [0, 1, 2, 3], 'pad_width': [-1, 1], 'mode': 'constant',
-     'constant_values': 3},
-    {'array': [0, 1, 2, 3], 'pad_width': [], 'mode': 'constant',
-     'constant_values': 3},
+     'kwargs': {'constant_values': 3}},
     {'array': [0, 1, 2, 3], 'pad_width': [[3, 4], [5, 6]], 'mode': 'constant',
-     'constant_values': 3},
+     'kwargs': {'constant_values': 3}},
     {'array': [0, 1, 2, 3], 'pad_width': [1], 'mode': 'constant',
-     'notallowedkeyword': 3},
-    # mode='edge'
-    {'array': [], 'pad_width': 1, 'mode': 'edge'},
-    {'array': [0, 1, 2, 3], 'pad_width': [-1, 1], 'mode': 'edge'},
-    {'array': [0, 1, 2, 3], 'pad_width': [], 'mode': 'edge'},
-    {'array': [0, 1, 2, 3], 'pad_width': [[3, 4], [5, 6]], 'mode': 'edge'},
+     'kwargs': {'notallowedkeyword': 3}},
+    # edge
+    {'array': [], 'pad_width': 1, 'mode': 'edge',
+     'kwargs': {}},
+    {'array': [0, 1, 2, 3], 'pad_width': [-1, 1], 'mode': 'edge',
+     'kwargs': {}},
+    {'array': [0, 1, 2, 3], 'pad_width': [[3, 4], [5, 6]], 'mode': 'edge',
+     'kwargs': {}},
     {'array': [0, 1, 2, 3], 'pad_width': [1], 'mode': 'edge',
-     'notallowedkeyword': 3},
+     'kwargs': {'notallowedkeyword': 3}},
     # mode='reflect'
-    {'array': [], 'pad_width': 1, 'mode': 'reflect'},
-    {'array': [0, 1, 2, 3], 'pad_width': [-1, 1], 'mode': 'reflect'},
-    {'array': [0, 1, 2, 3], 'pad_width': [], 'mode': 'reflect'},
-    {'array': [0, 1, 2, 3], 'pad_width': [[3, 4], [5, 6]], 'mode': 'reflect'},
+    {'array': [], 'pad_width': 1, 'mode': 'reflect',
+     'kwargs': {}},
+    {'array': [0, 1, 2, 3], 'pad_width': [-1, 1], 'mode': 'reflect',
+     'kwargs': {}},
+    {'array': [0, 1, 2, 3], 'pad_width': [[3, 4], [5, 6]], 'mode': 'reflect',
+     'kwargs': {}},
     {'array': [0, 1, 2, 3], 'pad_width': [1], 'mode': 'reflect',
-     'notallowedkeyword': 3},
+     'kwargs': {'notallowedkeyword': 3}},
 )
 @testing.gpu
-class TestPadFailure(unittest.TestCase):
+@testing.with_requires('numpy>=1.17')
```

Comment:
This test requires NumPy 1.17 or later.
In the previous `TestPadFailure` version, the test has not worked correctly because `AttributeError` was raised while accessing `self.constant_value`.